component/telemetry.go

```diff
@@ -30,10 +30,4 @@ type TelemetrySettings struct {
 
 	// Resource contains the resource attributes for the collector's telemetry.
 	Resource pcommon.Resource
-
-	// ReportStatus allows a component to report runtime changes in status. The service
-	// will automatically report status for a component during startup and shutdown. Components can
-	// use this method to report status after start and before shutdown. For more details about
-	// component status reporting see: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-status.md
-	ReportStatus func(*StatusEvent)
```

Comment:
I've changed my mind.  Keeping the `component` stuff around is very messy bc it expects to be using its types instead of `componentstatus`.  So `TelemetrySettings.ReportStatus` would use `component.StatusEvent` but `componentstatus.Reporter` uses `componentstatus.Event`.

I'll commit to updating jaeger once this breaking change is merged.  Luckily it is a one line change to `componentstatus.ReportStatus`, like in the otlpreceiver.

Reply:
@yuriolisa according to some Github Searches jaeger is the only external consumer of this API.  This PR is going to break:

- https://github.com/jaegertracing/jaeger/blob/main/pkg/telemetery/settings.go
- https://github.com/jaegertracing/jaeger/blob/main/cmd/collector/app/handler/otlp_receiver.go
- https://github.com/jaegertracing/jaeger/blob/main/cmd/remote-storage/app/server.go
- https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/integration/storagecleaner/extension.go
- https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/remotesampling/extension.go
- https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/server.go

The path forward for all scenarios is straightforward and I can submit the PR. The only interesting change would be to https://github.com/jaegertracing/jaeger/blob/main/pkg/telemetery/settings.go, as it would be a type change on a public field on the public struct. It looks like that struct was added very recently, so will changing it be ok? The alternative is to deprecate its `ReportStatus` and add a new one that uses `componentstatus.Event`.

Reply:
Yes sure, go ahead. It will actually be better to use componentstatus.Event over the previous implementation.